pkg/experiment/query_backend/query_profile_entry.go

```diff
@@ -93,6 +93,35 @@ func getSeriesLabels(reader phlaredb.IndexReader, matchers []*labels.Matcher, by
 	return series, postings.Err()
 }
 
+// getSeriesLabelsFingerprints is identical to getSeriesLabels,
+// but the returned map contains series fingerprints as keys,
+// instead of series indices.
+func getSeriesLabelsFingerprints(reader phlaredb.IndexReader, matchers []*labels.Matcher, by ...string) (map[uint64]seriesLabels, error) {
+	postings, err := getPostings(reader, matchers...)
+	if err != nil {
+		return nil, err
+	}
+	chunks := make([]index.ChunkMeta, 1)
+	series := make(map[uint64]seriesLabels)
+	l := make(phlaremodel.Labels, 0, 6)
+	for postings.Next() {
+		fp, err := reader.SeriesBy(postings.At(), &l, &chunks, by...)
+		if err != nil {
+			return nil, err
+		}
+		_, ok := series[fp]
+		if ok {
+			continue
+		}
```

Comment:
While it works, it's inefficient: we remember by series fingerprint, not the hash of the subset of labels – I'll fix it before merging